src/Symfony/Component/Security/Core/Authorization/Voter/CallableVoter.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+use Symfony\Component\Security\Core\Authentication\AuthenticationTrustResolverInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+final class CallableVoter implements CacheableVoterInterface
+{
+    public function __construct(
+        private AuthenticationTrustResolverInterface $trustResolver,
+        private AuthorizationCheckerInterface $authChecker,
+    ) {
+    }
+
+    public function supportsAttribute(string $attribute): bool
+    {
+        return false;
+    }
+
+    public function supportsType(string $subjectType): bool
+    {
+        return true;
+    }
+
+    public function vote(TokenInterface $token, mixed $subject, array $attributes): int
+    {
+        $result = VoterInterface::ACCESS_ABSTAIN;
+        foreach ($attributes as $attribute) {
+            if (!$attribute instanceof \Closure) {
+                continue;
+            }
+
+            $result = VoterInterface::ACCESS_DENIED;
+            if ($attribute($token, $subject, $this->trustResolver, $this->authChecker)) {
```

Comment:
should we pass the AuthenticationTrustResolverInterface at all ? We already have a voter allowing to handle trust checks through the AccessDecisionManagerInterface (with checks that are documented, unlike the lower-level TrustResolver API).
And at least, we should move it later in the argument list if we keep it, to make it easier to omit it for closures not needing it.

Reply:
It has some nice methods that could be useful, but I moved it to the last position indeed. It is less likely to be used than the other arguments.